tests/playgrounds/test_blocks_toolbox.js

```diff
@@ -889,35 +889,35 @@ var style = {
     },
     {
       "kind": "BLOCK",
-      "blockxml": "<block type=\"test_style_hat\"></block>",
+      "type": "test_style_hat",
     },
     {
       "kind": "LABEL",
       "text": "Colour"
     },
     {
       "kind": "BLOCK",
-      "blockxml": "<block type=\"test_style_hex1\"></block>",
+      "type": "test_style_hex1",
     },
     {
       "kind": "BLOCK",
-      "blockxml": "<block type=\"test_style_hex2\"></block>",
+      "type": "test_style_hex2",
     },
     {
       "kind": "BLOCK",
-      "blockxml": "<block type=\"test_style_hex3\"></block>",
+      "type": "test_style_hex3",
     },
     {
       "kind": "BLOCK",
-      "blockxml": "<block type=\"test_style_no_colour\"></block>",
+      "type": "test_style_no_colour",
     },
     {
       "kind": "BLOCK",
-      "blockxml": "<block type=\"test_style_hex4\"></block>",
+      "type": "test_style_hex4",
     },
     {
       "kind": "BLOCK",
-      "blockxml": "<block type=\"test_style_hex5\"></block>",
+      "type": "test_style_hex5",
```

Comment:
Based on the code, you can have blocks with both `type` and `gap`.  Add something to test that.